### DISCUSSION TITLE
State of BLE -> Zigbee now & best LCD sensor to consider

### DISCUSSION THREAD

**Discussion by mneveroff** [pvvx/ATC_MiThermometer#551] [Q&A] [answered]
Thanks @pvvx for your amazing work on this and accompanying firmware & tooling, it really did go a long way. I wanted, like many of us, to grab a few dozens of cheap BLE sensors to monitor conteiner humidity (for 3d printer filament storage), but reading over this and a few of your other repos it seems to be that the days of "buy -> OTA fw -> enjoy" are somewhat over? Would appreciate you or others chiming in.

1. By far the cheapest option is [Tuya TH05](https://www.aliexpress.com/item/1005006904475672.html) at ~$3, but following the findings on their [power draw](https://github.com/pvvx/THB2/issues/8#issuecomment-1908982171) and [overall quality](https://github.com/pvvx/THB2/issues/8#issuecomment-1909016545) I reckon I'd stay away. It also would require a [BLE -> Zigbee repeater](https://github.com/pvvx/TLB2Z), and that's just too many elements for me to go through a few dozens of.
2. The step-up and the most popular choice is [Xiaomi LYWSD03MMC](https://www.aliexpress.com/item/1005006861875198.html) at ~$5, but [ever since 2.1.1 release](https://github.com/atc1441/ATC_MiThermometer/issues/298) almost a year ago, the only way to get it to work is UART route which involves soldering and isn't as straightforward as it used to be. Best I understood, until Xiaomi releases an update over 2.1.1 there's no feasible path for the community to re-enable OTA, and they might as well just never release it.

All of the other options listed in [the repo](https://github.com/pvvx/ATC_MiThermometer) seem to climb to $10-15 and at that point lose the appeal as one could get a zigbee version of the either of the above and bypass the hassle.

Is my overall summary correct, and would that mean that if one wanted to get a few dozen hygrometers on the cheap (but not the refuse parts from Tuya) then the LYWSD03MMC + UART and downgrading firmware is the most sensible way to go time-wise?

**Comment by pvvx** [comment, marked as answer]
CR2032 batteries do not like high temperatures. At temperatures above +25C, self-discharge occurs...
As a result, using thermometers with CRxxxx elements in a filament dryer is not optimal.

> [Tuya TH05](https://www.aliexpress.com/item/1005006904475672.html) at ~$3,

Even cheaper are most [thermometers on PHY622x2 chips](https://github.com/pvvx/THB2) (on average ~$2).

> I reckon I'd stay away. It also would require a [BLE -> Zigbee repeater](https://github.com/pvvx/TLB2Z), and that's just too many elements for me to go through a few dozens of.

One USB-BT in Home Assistant works with more than fifty thermometers...
Any USB-BT adapter on the CSR8510A10 chip is suitable for this. Other USB-BT adapter options are slower and handle fewer BLE advertising packet receptions per second...

**Reply by mneveroff** [reply]
Yeah sorry didn't mean the actual dryers @pvvx, I figured the temp rather quickly (the sensors give out after +60C too, and dryers can comfortably go past 70) and have designed the dehydrator walls without any sensor cutouts precisely for that reason.

What I meant were not dryers but rather dryboxes, mybad: https://makerworld.com/en/models/34175?from=search#profileId-31358. So those would be room temp at most. Basically recreating something like this at 1/5 the cost: https://www.reddit.com/r/BambuLab/comments/1fktu2z/comment/lnyfn4y/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

> Even cheaper are most [thermometers on PHY622x2 chips](https://github.com/pvvx/THB2) (on average ~$2).

Yeah, that's exactly what I meant under Tuya devices, but the comment about them using spare (negative cost) parts and V1.4 being utterly stupid for performance puts me off. Also, wouldn't all devices from that repo require a BLE -> Zigbee repeater, making it a much more time-consuming endavour, or am I mistaking it for something?

That's the reason why I felt going with LYWSD03MMC might be sensible - each will be it's own ZigBee thingy, requiring a quick solder and fw update, or is there a better / more cost-effort efficient way (taking into consideration time/cost for repeaters with the PHY622x2 way which are all Tuya IIRC)?

**Comment by pvvx** [comment]
>> I reckon I'd stay away. It also would require a [BLE -> Zigbee repeater](https://github.com/pvvx/TLB2Z), and that's just too many elements for me to go through a few dozens of.

> One USB-BT in Home Assistant works with more than fifty thermometers...
Any USB-BT adapter on the CSR8510A10 chip is suitable for this. Other USB-BT adapter options are slower and handle fewer BLE advertising packet receptions per second...

Currently, 37 BLE devices are stably received on one USB-BT adapter (CSR8510A10):
![image](https://github.com/user-attachments/assets/b3d8ba6e-0b80-4297-9636-e2d368d798b0)

Some of the thermometers:
![image](https://github.com/user-attachments/assets/1ec93f25-7916-4140-a9bb-62eab8d3c919)

I have 17 more BLE devices on the second USB-BT, but they are very remote (working in "LE Long Range" mode). Zigbee or WiFi do not reach them. Therefore, BLE is used in "LE Long Range" mode.

**Reply by mneveroff** [reply]
Oh wow, this somehow entirely slipped past me, but this by far would be the most cost-effective way of doing that, thank you! And mostly hassle-free I imagine, with the only caveat of TH05_V1.4 being incredibly power-hungry if I'm unlucky with the components?

I was under the impression that one could only pair three BLE devices to a single repeater, hence the concern (https://github.com/pvvx/TLB2Z), but now I realise that it's only true for BT->ZigBee, but just grabbing a USB-BT one is likely a much better option indeed. Or I can just pair them to the HA on RPI via BT directly I reckon. Thanks for that shout, I think I'll go that route. My only concern is power draw, since those batteries costs do add up, is it down to the Aliexpress lottery if I get [the CHT8305](https://github.com/pvvx/THB2/issues/8#issuecomment-1908982171) or not?

Thank you so much anyways, this has informed my planning a whole lot better already!
